pkg/controller/replica.go

```diff
@@ -108,13 +108,25 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 			if !isCleanupRequested(azVolumeAttachment) {
 				go r.handleReplicaDelete(context.Background(), azVolumeAttachment)
 			}
-			// if replica attachment failed, delete it and create a replacement
 		} else if azVolumeAttachment.Status.State == azdiskv1beta2.AttachmentFailed {
-			// if attachment failed for replica AzVolumeAttachment, delete the CRI so that replace replica AzVolumeAttachment can be created.
-			if err := r.cachedClient.Delete(ctx, azVolumeAttachment); err != nil {
-				return reconcile.Result{Requeue: true}, err
+			// if the failed attachment is allowed to retry, reset the state to pending
+			if azureutils.MapContains(azVolumeAttachment.Status.Annotations, consts.VolumeAttachRetryAnnotation) {
+				updateFunc := func(obj client.Object) error {
+					azva := obj.(*azdiskv1beta2.AzVolumeAttachment)
```

Comment:
Let's remove the `consts.VolumeAttachRetryAnnotation` annotation here when we reset the state to `Pending` because we won't be needing it after that.